File: src/Core/Houston.Application/ViewModel/ConnectorFunctionHistoryViewModels/ConnectorFunctionHistoryDetailViewModel.cs
namespace Houston.Application.ViewModel.ConnectorFunctionHistoryViewModels {
	public class ConnectorFunctionHistoryDetailViewModel {
		public Guid Id { get; set; }

		public Guid ConnectorFunctionId { get; set; }

		public string Version { get; set; } = null!;

		public byte[] Script { get; set; } = null!;

		public byte[] Package { get; set; } = null!;

		public List<ConnectorFunctionInputViewModel> Inputs { get; set; } = new List<ConnectorFunctionInputViewModel>();

		public BuildStatus BuildStatus { get; set; }

		public byte[]? BuildStderr { get; set; }

		public string CreatedBy { get; set; } = null!;

		public DateTime CreationDate { get; set; }

		public string UpdatedBy { get; set; } = null!;

		public DateTime LastUpdate { get; set; }
	}
}

File: src/Web/WebAngular/src/app/domain/view-models/connector-function-history-detail.view-model.ts
import { BuildStatus } from "../enums/build-status.enum";
import { ConnectorFunctionInputViewModel } from "./connector-function-input.view-model";

export interface ConnectorFunctionHistoryDetailViewModel {
	id: string;
	connectorFunctionId: string;
	version: string;
	script: Buffer | Uint8Array | string;
	package: Buffer | Uint8Array | string;
	inputs: ConnectorFunctionInputViewModel[];
	buildStatus: BuildStatus;
	buildStderr: Buffer | Uint8Array | string | null;
	createdBy: string;
	creationDate: string;
	updatedBy: string;
	lastUpdate: string;
}
